build-windows.ps1

```diff
@@ -60,28 +60,36 @@ if ($Debug) {
 
 $cmakeHelp=$(cmake --help)
 
-# -requires param : ensure that the visual studio installs have the C++ workload
```

Comment:
I'm not sure what the purpose of this change to generator selection is. The intent of my ask for the PR was to add a line for support of VS2022 to the enum switch.

Perhaps this was done unintentionally, but it looks like the check for `Microsoft.VisualStudio.Component.VC.Tools.x86.x64` has been removed. This is important since some people have VS installed without the C++ tools. It shouldn't try to build with VS if the C++ tools aren't installed, that's the same as not having VS installed at all since the compiler toolchain is missing.

Unless there's a compelling reason for this change I'm missing, please back it out and just add the line for VS2022 that you've been recommending in 404 chat.